crates/common/src/lib.rs

```diff
@@ -144,6 +144,10 @@ macros::i64_backed_u64::to_from_sql!(StarknetBlockTimestamp);
 macros::i64_backed_u64::new_get_partialeq!(StarknetBlockTimestamp);
 macros::i64_backed_u64::serdes!(StarknetBlockTimestamp);
 
+/// A StarkNet block commitment.
+#[derive(Copy, Clone, PartialEq, Eq, Deserialize, Serialize)]
+pub struct StarknetCommitment(pub StarkHash);
```

Comment:
I think it would make sense to create two separate types here, `EventCommitment` and `TransactionCommitment`.